pkg/controller/supportbundlecollection/controller.go

```diff
@@ -627,6 +677,175 @@ func (c *Controller) isCollectionAvailable(bundleCollection *v1alpha1.SupportBun
 	return true
 }
 
+func (c *Controller) updateStatus(internalBundleCollection *types.SupportBundleCollection) error {
+	updateStatusFunc := func(currentNodes, desiredNodes int, updatedConditions []v1alpha1.SupportBundleCollectionCondition) error {
+		status := &v1alpha1.SupportBundleCollectionStatus{
+			SucceededNodes: int32(currentNodes),
+			DesiredNodes:   int32(desiredNodes),
+			Conditions:     updatedConditions,
+		}
+		klog.V(2).InfoS("Updating SupportBundleCollection status", "supportBundleCollection", internalBundleCollection.Name, "status", status)
+		return c.updateSupportBundleCollectionStatus(internalBundleCollection.Name, status)
+	}
+
+	// It means the SupportBundleCollection hasn't been processed once. Set it to started failed to differentiate from
+	// SupportBundleCollection that spans 0 Node.
+	now := metav1.Now()
+	if internalBundleCollection.SpanMeta.NodeNames == nil {
+		return updateStatusFunc(0, 0, []v1alpha1.SupportBundleCollectionCondition{
+			{
+				Type:               v1alpha1.CollectionStarted,
+				Status:             metav1.ConditionFalse,
+				LastTransitionTime: now,
+			},
+		})
+	}
+
+	desiredNodes := len(internalBundleCollection.SpanMeta.NodeNames)
+	succeededNodes := 0
+	// TODO: Use a map to store the failed Node/ExternalNode and its error, and report the failed list when the
+	// collection is completed.
+	failedNodes := make(map[string][]string)
+	failedNodesCount := 0
```

Comment:
can we name counts consistently? desiredNodes, succeededNodes, failedNodes, failedNodeReasons?

Reply:
Renamed the variables accordingly.